dataplicity/client.py

```diff
@@ -95,31 +96,48 @@ def disk_poll(self):
 
         if now >= self.next_disk_poll_time:
             self.next_disk_poll_time = now + self.disk_poll_rate_seconds
-            disk_space = disk_usage('/')
+            disk_space = disk_usage("/")
 
             with self.remote.batch() as batch:
                 batch.call_with_id(
-                    'authenticate_result',
-                    'device.check_auth',
-                    device_class='tuxtunnel',
+                    "authenticate_result",
+                    "device.check_auth",
+                    device_class="tuxtunnel",
                     serial=self.serial,
-                    auth_token=self.auth_token
+                    auth_token=self.auth_token,
                 )
                 batch.call_with_id(
-                    'set_disk_space_result',
-                    'device.set_disk_space',
+                    "set_disk_space_result",
+                    "device.set_disk_space",
                     disk_capacity=disk_space.total,
-                    disk_used=disk_space.used
+                    disk_used=disk_space.used,
                 )
 
+    def tag_poll(self):
+        with self.remote.batch() as batch:
```

Comment:
Can we avoid making this call if the tag list hasn't changed since the last successful request? Remember we have may 1000s of devices. Best to avoid them making requests if we can avoid it...

Reply:
You don't need to be too persistent about it.  An in memory comparison would be fine.  If one call is always made after restarting the agent, so be it.  But that's the 1% case, @willmcgugan is right on the 99% case here.